main/webapp/modules/core/scripts/project.js

```diff
@@ -105,6 +105,7 @@ function resizeAll() {
   ui.processPanel.resize();
   ui.historyPanel.resize();
   ui.dataTableView.resize();
+  SchemaAlignment.resize();
```

Comment:
This `SchemaAlignment` is coming from the Wikidata/Wikibase extension, so it might not be there if OpenRefine is running without this extension. I wonder if we have any extension point for this, or if we should build one?

I assume that this `resize()` business should really not be there in the first place, and we should just let the browser do its job of figuring out the layout, given sensible CSS directives. But changing to something else is not going to happen overnight.

Reply:
There already is an API: the `window.resize` event handler.  I tried registering the `SchemaAlignment.resize()` independently, but it didn't seem to work.  I think there is some dependency on the order of the other resize code in `resizeAll()` and then `SchemaAlignment.resize()`.

I can put a guard to check if `SchemaAlignment` is defined in case the extension is missing.

There is quite a bit of work to untangle the Wikidata UI from the rest of the UI, that is beyond the scope of this PR.

Reply:
> There is quite a bit of work to untangle the Wikidata UI from the rest of the UI, that is beyond the scope of this PR.

Where do you see other coupling points? I am not aware of anywhere else in the main code that refers to the Wikidata extension. The Wikidata extension inserts its Schema, Issues and Preview tabs in a hacky way, without having a corresponding extension point in the core, but that is not a dependency from the core to the extension.

I would rather try to introduce an extension point which lets extensions define new tabs in the main panel. The logic of switching between tabs would be handled by the core, and so would be the resizing.

Reply:
Regarding the coupling points, I was just talking about how the sizing of the Wikidata tabs/panels is coupled to the resizing of the right-hand-side view panel somehow.  I tried decoupling by having Wikidata extension separately subscribe to `window.resize` event but that didn't work.

This PR is just trying to fix the issue, the coupling was already present. If you look at `switchTab()`, it was already calling `resizeAll()` in projects.js.  I just corrected the logic so it is happening on resize also.

If/when the Wikidata extension will be extracted into a separate repo, the decoupling work can be taken up as part of that.

Also my IDE says there are 27 dependencies from just wikidata/schema-alignment.js to the main webapp, pictured in the screenshot.  I haven't looked at the extension APIs, so not sure how many of these dependencies are using official extension API, versus private/internal APIs/variables.   That is just one file of Wikidata extensions. There are probably 100s of dependencies in the whole extension.

RHS: schema-alignment.js   LHS: main webapp scope
![image](https://user-images.githubusercontent.com/42903164/195511211-afd29975-9990-48d1-8f3a-ab2177c69cd7.png)

Reply:
> Also my IDE says there are 27 dependencies from just wikidata/schema-alignment.js to the main webapp, pictured in the screenshot.

Sure, in this direction it is totally expected.

>  I haven't looked at the extension APIs, so not sure how many of these dependencies are using official extension API, versus private/internal APIs/variables.

I think we have just not documented at all what the frontend extension points are meant to be, so it's hard to tell!